adding in operatorhub onboarding documents (#1349)
aws-controllers-k8s/community: docs/content/docs/contributor-docs/operatorhub-onboarding.md

---
title: "OperatorHub Onboarding"
description: "How controllers end up in OperatorHub"
lead: ""
draft: false
menu:
  docs:
    parent: "contributor"
weight: 20
toc: true
---

There are two ways a user can install an operator, one is via an OLM enabled cluster using OperatorHub.io and the other

> is via the embedded OperatorHub within an OpenShift cluster. Inorder to onboard a new controller and have it appear in both

Reviewer: ```suggestion
is via the embedded OperatorHub within an OpenShift cluster. In order to onboard a new controller and have it appear in both
```

places, the below steps should be followed. After these steps are completed, the build/release process will then raise pull
requests against the proper repos.

## Add an OLM Config File to Controller Repository

> This file is used during the build/release process of a controller to assist in the generation of the CSV in the controllers bundle.

Reviewer 2: ```suggestion
This file is used during the build/release process of a controller to assist in the generation of the CSV in the controller's bundle.
```

Author: fixed

> This file is used during the build/release process of a controller to assist in the generation of the CSV in the controllers bundle.

Reviewer 2: ```suggestion
This file is used during the build/release process of a controller to assist in the generation of the [ClusterServiceVersion](https://olm.operatorframework.io/docs/concepts/crds/clusterserviceversion/) ("CSV") in the controllers bundle.
```

Author: fixed

The file should live at `./olm/olmconfig.yaml` in the project structure of a controller. It should also contain a sample for each `CustomResource`

> manged by the controller. Please see the S3 controllers `olmconif.yaml` found

Reviewer: ```suggestion
manged by the controller. Please see the S3 controllers `olmconfig.yaml` found
```

> manged by the controller. Please see the S3 controllers `olmconif.yaml` found

Reviewer 2: ```suggestion
managed by the controller. Please see the S3 controllers `olmconif.yaml` found
```

> manged by the controller. Please see the S3 controllers `olmconif.yaml` found

Author: fixed

## Validate the Generated CSV

After the `olmconfig.yaml` has been generated it's a good practice to validate that the CSV for the controller looks
as expected, this ensures proper AWS branding for the controller. This does not need to be done after ever change to the
controller, but if changes to the `olmconfig.yaml` are done, or a new CR is added to the controller, it's best to validate that
the new changes in the CSV appears as expected.

1. Build the controller locally using the `code-generator` project
   1. Install Operator SDK in the `code-generator` `/bin` directory using the below script
      1. `$ ./scripts/install-operator-sdk.sh`
   2. Target the appropriate controller
      1. `$ export SERVICE=s3`
   3. Build the controller and generate the bundle
      1. `$ ACK_GENERATE_OLM=true make build-controller SERVICE=$SERVICE`
2. Validate that the CSV was generated
   1. Unless overridden, the CSV will be at `$GOPATH/src/github.com/aws-controllers-k8s/s3-controller/olm/bundle/manifests/ack-s3-controller.clusterserviceversion.yaml`
3. The CSV can be [previewed](https://operatorhub.io/preview) by coping and pasting the CSV

>    ![OperatorHub.io Preview](../images/operatorhub-preview.png)

Reviewer: This image is super wide. Would you mind cropping down the whitespace on the outsides? I think we have something like 1024px at most for the content

Author: - Relates: #1354 

The above PR should help resolve this issue, since the formatting was only in the dark style that was limiting image widths.

## Raise Pull Requests to Community Operators Repositories

> Both repositories rely on the same folder structure for each operator, which is laid out below. For the initial onboarding

Author: I copied from both repos readme's, let me know if this is helpful or more confusing.

all that needs to be worried about is adding `./operators/ack-new-controller` and the `ci.yaml` file. Since the ACK project

> release all the operators using semver versioning, each ack operator CI file will be identical, so and exiting ACK operators

Reviewer 2: ```suggestion
release all the operators using semantic versioning, each ack operator CI file will be identical, so and exiting ACK operators
```

> release all the operators using semver versioning, each ack operator CI file will be identical, so and exiting ACK operators

Reviewer 3: ```suggestion
releases all the operators using semver versioning, each ACK operator CI file will be identical, so an existing ACK operator's
```
